### ISSUE TITLE
Support retrieving binary resources from Config Server

### ISSUE THREAD

**Issue by Walliee** [pivotal-cf/spring-cloud-services-connector#120]
Config Server allows fetching resources in both plaintext and binary form.
spring-cloud-services-connector supports fetching plaintext resources but lacks support for binary resources. This will be useful for fetching binary files like images. 

**Implementation Notes**
There is an existing client interface for fetching plaintext files from Config Server called `PlainTextConfigClient`. I propose refactoring this interface to be called `ConfigResourceClient` and it's existing methods `getConfigFile()` to be `getPlainTextResource()`.

New `getBinaryResource()` methods can then be added to support retrieving binary resource.

I can work on  a PR if there is consensus that the proposed changes make sense.

**Comment by Walliee** [comment]
CC: @daniellavoie as you originally added support for fetching plaintext resources.

**Comment by Walliee** [comment]
I did some work on this here: https://github.com/pivotal-cf/spring-cloud-services-connector/compare/master...Walliee:GH-120

Please let me know if this looks good and I can open a PR.

**Comment by csterwa** [comment]
@Walliee We appreciate the work you've done. Due to a release that we are focused on over the next few weeks we won't be able to merge this until then. There is some testing and in-depth review that we  will do at that time. Are you able to use this off of your branch until that time?

**Comment by Walliee** [comment]
I can't really use it off of my personal branch at work but as this is an enhancement request it sounds totally reasonable to me that it be included in the next release cycle.

**Comment by csterwa** [comment]
Thank you @Walliee - we'll work on getting it prioritized just after our release.

**Comment by Walliee** [comment]
Hey @habuma! I opened a PR for this. #122 
Let me know your thoughts.

**Comment by habuma** [comment]
@walliee Yes, I saw your PR and thanks! But I saw it *AFTER* having spent some time reviewing your branch and reworking your stuff a bit. We'd like to include this in a minor release, but renaming types and methods breaks backward compatibility and thus breaks semantic versioning if we release it in a minor release. So, I used your work as a starting point and changed a few things to avoid renaming anything. I've tried to retain `@author` credit in the JavaDoc for you, but if I left you out of anything, let me know, as I want to make sure you get credit for the work you did.

It's in the `master` branch now if you want to give it a spin and give feedback.

**Comment by spikymonkey** [comment]
@habuma I can confirm this works. I knocked together a little WebMVC app with a request mapping like this:

```
    @GetMapping(value = "/binary/{filename}")
    public ResponseEntity<Resource> binaryResource(@PathVariable String filename) {
        Resource resource = configResourceClient.getBinaryResource("cloud", "master", filename);
        HttpHeaders headers = new HttpHeaders();
        headers.setContentType(MediaType.APPLICATION_OCTET_STREAM);
        return new ResponseEntity<>(resource, headers, HttpStatus.OK);
    }
```

then bound this app to a config server instance targeting a Git repo with a file called `image.jpeg` at the root. I was then able to hit `my-app-url/binary/image.jpeg` and retrieve the file through Chrome. It was indeed still a picture of a cat, so the bits didn't get scrambled along the way :)

**Comment by Walliee** [comment]
@habuma No worries, thanks for looking into it. LGTM!

closed #122
